### PR TITLE
applying to staff.   staff view with approved and unapproved list

### PR DESCRIPTION
**Pull request by SSESoup**
So it's not complete but here it is today.  Sorry for not pushing this earlier I was out and did not see the msg for pull request.  there will prob be merge conflicts with master but for now here it is to review.   
Still needs:
-needs to allow approving as staff admin.
-adding client admins.  
-haven't ran flake8 yet so I will fix that when I fix the merge conflicts

### CODE REVIEW

File: timetracker/vms/templates/vms/dashboard.html [hunk 1]
```diff
@@ -4,13 +4,14 @@
   <h1>Dashboard</h1>
   <h2>Quick Links</h2>
   {% if request.user.is_staff %}
-    <a href="{% url 'vms:create-staff-agency' %}">create a staffing agency!</a>
+    <a href="{% url 'vms:create-staff-agency' %}">create a staffing agency!</a><br>
```
Comment (cdriehuys) [inline timetracker/vms/templates/vms/dashboard.html]: Only superusers are able to create staffing agencies, and that will be done from the admin interface as we discussed with Nick, so this link is not needed.

File: timetracker/vms/templates/vms/dashboard.html [hunk 2]
```diff
@@ -4,13 +4,14 @@
...
   {% endif %}
+  <a href="{% url 'vms:create-staff-employee' %}">apply as a staffing agency employee!</a><br>
   {% if staff_admin %}
-    <h2>Staff List</h2>
+    <h2>Staff</h2>
     <ul>
         {% for staff in staff_admin %}
-          <li>{{ staff.agency }}</li>
+          <li><a href= "{{ staff.agency.get_absolute_url }}">{{ staff.agency }}</a></li>
```
Comment (cdriehuys) [inline timetracker/vms/templates/vms/dashboard.html]: The changes to the listing of staffing agencies the user is an admin for have already been made in the master branch, so look out for that when rebasing.

File: timetracker/vms/forms.py [hunk 3]
```diff
@@ -130,6 +131,27 @@ def save(self, user):
             agency=agency
         )
 
+class CreateStaffEmployeeForm(forms.Form):
+    """
+    Form to create staffing employee.
+    """
+    agency_field=forms.ModelChoiceField(required=True,
```
Comment (cdriehuys) [inline timetracker/vms/forms.py]: By default the name of the field is shown to the user, so if the name is something like `*_field`, add a `label=<human friendly name>` to the field parameters.

File: timetracker/vms/forms.py [hunk 4]
```diff
@@ -130,6 +131,27 @@ def save(self, user):
...
+    queryset=models.StaffingAgency.objects.all(),
+    empty_label=None)
+
+    def save(self, user):
+        agency = self.cleaned_data['agency_field']
+        agency_object=models.StaffingAgency.objects.get(name=agency)
```
Comment (cdriehuys) [inline timetracker/vms/forms.py]: There's no need for this. The `ModelChoiceField` returns an instance of the model already, so `self.cleaned_data['agency_field']` *is* a `StaffingAgency` instance.

**Comment by SSESoup** [comment]
okay I got rid of the already implemented stuff.  sorry for not pushing this code freq my masters was quite behind.  it's basically only has the form to applying as a staffing employee now

**Pull request merged**